libs/elasticsearch-nio/src/main/java/org/elasticsearch/nio/RoundRobinSupplier.java

```diff
@@ -19,21 +19,22 @@
 
 package org.elasticsearch.nio;
 
+import java.util.List;
 import java.util.concurrent.atomic.AtomicInteger;
 import java.util.function.Supplier;
 
 public class RoundRobinSupplier<S> implements Supplier<S> {
 
-    private final S[] selectors;
+    private final List<S> selectors;
     private final int count;
     private AtomicInteger counter = new AtomicInteger(0);
 
-    RoundRobinSupplier(S[] selectors) {
-        this.count = selectors.length;
+    RoundRobinSupplier(List<S> selectors, int count) {
```

Comment:
I am not sure we should use the List interface if we access stuff randomly. In the case of a linked list access will be linear. We can just call toArray if that's easiest?!